doc/dev/developer_guide/basic-workflow.rst

```diff
@@ -29,109 +29,161 @@ The following chart illustrates the basic Ceph development workflow:
 
 See the `GitHub documentation
 <https://help.github.com/articles/fork-a-repo/#platform-linux>`_ for
 detailed instructions on forking. In short, if your GitHub username is
 "mygithubaccount", your fork of the upstream repo will appear at
 ``https://github.com/mygithubaccount/ceph``. 
 
+.. _cloning:
+
 Cloning Your Fork  
 ^^^^^^^^^^^^^^^^^
 
-Once you have created your fork, clone it by running:
+After you have created your fork, clone it by running the following command:
 
 .. prompt:: bash $
 
    git clone https://github.com/mygithubaccount/ceph
 
-You must fork the Ceph repository before you clone it.  Without forking, you cannot 
-open a `GitHub pull request
+You must fork the Ceph repository before you clone it.  If you fail to fork,
+you cannot open a `GitHub pull request
 <https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_.
 
 For more information on using GitHub, refer to `GitHub Help
 <https://help.github.com/>`_.
 
 Configuring Your Local Environment
-----------------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-In the local environment created in the previous step, you now have a copy of
-the ``master`` branch in ``remotes/origin/master``. This fork
-(https://github.com/mygithubaccount/ceph.git) is frozen in time and the
-upstream repo (https://github.com/ceph/ceph.git, typically abbreviated to
-``ceph/ceph.git``) is updated frequently by other contributors. This means that
-you must sync your fork periodically. Failure to synchronize your fork may
-result in your commits and pull requests failing to merge because they refer to
-file contents that have changed since you last synchronized your fork.
+The commands in this section configure your local git environment so that it
+generates "Signed-off-by:" tags. They also set up your local environment so
+that it can stay synchronized with the upstream repository.
 
-Configure your local git environment with your name and email address.  
+These commands are necessary only during the initial setup of your local
+working copy. Another way to say that is "These commands are necessary
+only the first time that you are working with the Ceph repository. They are,
+however, unavoidable, and if you fail to run them then you will not be able
+to work on the Ceph repository.".
 
-.. prompt:: bash $
+1. Configure your local git environment with your name and email address.  
 
-   git config user.name "FIRST_NAME LAST_NAME"
-   git config user.email "MY_NAME@example.com"
+   .. prompt:: bash $
 
-Add the upstream repo as a "remote" and fetch it:
+      git config user.name "FIRST_NAME LAST_NAME"
+      git config user.email "MY_NAME@example.com"
 
-.. prompt:: bash $
+2. Add the upstream repo as a "remote" and fetch it:
 
-   git remote add ceph https://github.com/ceph/ceph.git
-   git fetch ceph
+   .. prompt:: bash $
+
+      git remote add ceph https://github.com/ceph/ceph.git
+      git fetch ceph
+
```

Comment:
Some other git resources name the upstream repository literally "upstream", which is what I do.  I suggest that following that convention here might make the process less confusing than "ceph" which is used pretty much everywhere.

Reply:
In my heart, I agree with you. But this naming convention is referred to in the Sepia wiki and in the mailing lists, so I'm chary of messing with it.

If we do decide to make this more conventional by using "upstream", I'd like to do it in a separate bug that touches only this issue in this file. I also imagine that such a thing will be part of a tracker-ticket-governed multiple-file-touching hunt for needles in haystacks.